internal/telemetry/client.go

```diff
@@ -389,48 +416,86 @@ func (c *Client) newRequest(t RequestType) *Request {
 			LanguageVersion: runtime.Version(),
 		},
 		Host: Host{
-			Hostname:    hostname,
-			ContainerID: internal.ContainerID(),
-			OS:          getOSName(),
-			OSVersion:   getOSVersion(),
+			Hostname:  hostname,
+			OS:        getOSName(),
+			OSVersion: getOSVersion(),
+			// TODO (lievan): arch, kernel stuff?
 		},
 	}
+	header := &http.Header{
+		"DD-API-KEY":                 {c.APIKey}, // DD-API-KEY is required as of v2
```

Comment:
Update to only send the api key _if_ we have one **and** we're talking directly to telemetry intake